app/models/vm_or_template.rb

```diff
@@ -1881,10 +1881,16 @@ def self.includes_template?(ids)
 
   def self.cloneable?(ids)
     vms = VmOrTemplate.where(:id => ids)
-    return false if vms.blank?
+    return false if vms.blank? || vms.any? {|vm| vm.orphaned? || vm.archived? }
     vms.all?(&:cloneable?)
   end
 
+  def self.publishable?(ids)
+    vms = VmOrTemplate.where(:id => ids)
```

Comment:
Let's make the method work on the collection directly, and leave the `where(:id)` to its caller.

Then we can just use `all` in here, and call it as `vms.publishable?`.